### PR TITLE
[FLINK-7118] [hadoop] Remove hadoop1.x code in HadoopUtils

### PR DESCRIPTION
**Pull request by zhangminglei**

### CODE REVIEW

**Comment by StephanEwen** [comment]
Looks good, thanks, merging this...

**Comment by StephanEwen** [comment]
Actually, sorry, is it possible to replace the code with just a proper `new TaskAttemptContextImpl(...)` call, rather than using reflection?

**Comment by zhangminglei** [comment]
Thanks, Stephan. I think it is a good idea. PR has been updated. :)

**Comment by StephanEwen** [comment]
Since these utility methods are now so simple, I think it makes sense to inline them in the two places where they are called. Then we could also get rid of the extra exception catch blocks and avoid the extra wrapping into RuntimeExceptions.

**Comment by zhangminglei** [comment]
@StephanEwen PR have been updated. Please check it out again ~

File: flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java [hunk 1]
```diff
@@ -126,18 +128,18 @@ public void open(int taskNumber, int numTasks) throws IOException {
 			this.jobConf.setInt("mapreduce.task.partition", taskNumber + 1);
 
 			try {
-				this.context = HadoopUtils.instantiateTaskAttemptContext(this.jobConf, taskAttemptID);
+				this.context = new TaskAttemptContextImpl(this.jobConf, taskAttemptID);
 			} catch (Exception e) {
-				throw new RuntimeException(e);
+				throw new IOException("Could not create instance of TaskAttemptContext.", e);
```
Comment (StephanEwen) [inline flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java]: I think the `TaskAttemptContextImpl` construction does not throw any Exception, so we can drop the try/catch block around it.

File: flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java [hunk 2]
```diff
@@ -126,18 +128,18 @@ public void open(int taskNumber, int numTasks) throws IOException {
...
 
 			this.outputCommitter = this.jobConf.getOutputCommitter();
 
 			JobContext jobContext;
 			try {
-				jobContext = HadoopUtils.instantiateJobContext(this.jobConf, new JobID());
+				jobContext = new JobContextImpl(this.jobConf, new JobID());
 			} catch (Exception e) {
-				throw new RuntimeException(e);
+				throw new IOException("Could not create instance of JobContext.", e);
```
Comment (StephanEwen) [inline flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java]: Same as above...

File: flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java (hunk 1 above)
Comment (zhangminglei) [inline flink-java/src/main/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatBase.java]: Yes. Stephan is correct!

**Comment by tillrohrmann** [comment]
Thanks for your contribution @zhangminglei. Changes look good to me. Merging this PR together with #4362.

**Pull request closed without merging**